### PR TITLE
Herwig7.1.2 Interface in CMSSW 

### PR DESCRIPTION
**Pull request by Andrej-CMS**
https://github.com/cms-sw/cmssw/pull/21050

### CODE REVIEW

**Comment by perrozzi** [comment]
please test with #21052 cms-sw/cmsdist#3519

**Comment by perrozzi** [comment]
+1

**Comment by smuzaffar** [comment]
please test with #21052 cms-sw/cmsdist#3642
cms-sw/cmsdist#3519 was for 94X, although test worked fine but let retry with correct branch.

**Comment by perrozzi** [comment]
+1

**Comment by fabiocos** [comment]
This looks an entire new package, for which there is no dedicated test workflow (as discussed in 2018/1/9 ORP meeting). 
1) A RelVal validation test to prove that the code really works is needed. There is a test workflow available in the test directory, could you use it as a basis for this?
2) what is the relationship and dependency of this PR with 21052? Do I correctly understand that that workflow suppresses the whole ThePergInterface package? Are they mutually incompatible, taking into account external update?
3) the test above shows CMSSW code rules problems, could you please check?

**Comment by Andrej-CMS** [comment]
@fabiocos 
1. I just added the RelVal workflow for QCD with H7
2. ThePEGInterface and H7 Interface use the same namespace. It was decided to drop ThePEGInterface, since H7 is the successor of Herwigpp
3. Could you please point me to the errors?

**Comment by perrozzi** [comment]
Hi Andrej, Fabio is referring to
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21711/25170/codeRules/cmsCodeRule3.txt
in the meanwhile I will trigger again the tests to check also the workflow, thanks a lot

**Comment by perrozzi** [comment]
please test with workflow 511.0  #21052 cms-sw/cmsdist#3642

**Comment by perrozzi** [comment]
please test workflow 511.0 with #21052 cms-sw/cmsdist#3642

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -909,6 +909,7 @@ def identityFS(wf):
 steps['SoftQCDinelastic_13TeV_pythia8']=genvalid('SoftQCDinelastic_13TeV_pythia8_cff',step1GenDefaults)
 
 steps['QCD_Pt-30_8TeV_herwigpp']=genvalid('QCD_Pt_30_8TeV_herwigpp_cff',step1GenDefaults)
```
Comment (fabozzi) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: Since you have updated the wf 511, the old step could be removed as well

**Comment by fabiocos** [comment]
please test workflow 511.0 with #21052 cms-sw/cmsdist#3642

**Comment by fabiocos** [comment]
Apparently the test failure is unrelated to the wf 511 (problem with 10224 DAS timeout noticed in IB already)

**Comment by Andrej-CMS** [comment]
Dear @all,
please test again, maybe DAS timeout will not happen this time.

In regards of the code rules: It seems like it complains about the usage of catch (...) . So, should I remove the error handling for unknown errors? (I would rather not remove it)

Kind regards,
Andrej

**Comment by ghellwig** [comment]
please test

**Comment by ghellwig** [comment]
@Andrej-CMS a catch-all is inefficient and unknown exceptions are handled by the framework.
In general, it is cleaner to explicitely write down the list of exceptions that are expected instead of just swallowing everything without giving much detail.

**Comment by ghellwig** [comment]
please test workflow 511.0 with #21052 cms-sw/cmsdist#3642

**Comment by ghellwig** [comment]
sorry, my bad

**Comment by Andrej-CMS** [comment]
The error above seems to result from the build not using the correct cmsdist version which includes the Herwig7 updates.

I have a short question: Is there any time constraint on the RelVals workflow? It takes some time to compile and integrate the matrix elements (>2h)

**Comment by perrozzi** [comment]
please test workflow 511.0 with #21052 cms-sw/cmsdist#3642

**Comment by fabiocos** [comment]
please test workflow 511.0 with #21052 cms-sw/cmsdist#3642

**Comment by fabiocos** [comment]
@Andrej-CMS if the problem is a too long calculation, why don't you select a simpler LO setup for this RelVal? This is first of all a basic technical test to verify that the basic external integration and interface are operational. More sophisticated checks should be done in a dedicated setup.

**Comment by fabiocos** [comment]
@Andrej-CMS BTW, is this test smoothly running in a private build, if you have one?

**Comment by Andrej-CMS** [comment]
@fabiocos 

I added LO Madgraph+Openloops+Herwig7 QCD as process. The work flow takes roughly one hour locally

**Comment by fabiocos** [comment]
please test workflow 511.0 with #21052 cms-sw/cmsdist#3642

**Comment by perrozzi** [comment]
+1

**Comment by fabiocos** [comment]
@kpedro88, @fabozzi this PR requires your signature after (successful) test, could you please provide it? As far as I can see we can integrate the new generator in pre4 for more extensive technical and physics tests. Otherwise please formulate your remarks.

**Comment by fabozzi** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**